apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -233,8 +233,7 @@ public override int GetHashCode()
         {
             unchecked
             {
-                int hashCode = StructMembers?.GetHashCode() ?? 0;
-                hashCode = (hashCode * 397) ^ (int) TypeCode;
+                int hashCode = (int) TypeCode;
```

Comment:
two objects can have the same hashcode yet not be equal, but if they are equal, their hashcodes must be the same.
The problem is the struct dictionary -- where I am manually checking for equality by scanning contents, but was incorrectly assuming the hashcodes of two dictionaries would be the same if their contents were the same.
This would decrease perf slightly for dictionaries keyed off of SpannerDbType that had a lot of structs (due to having many colliding hashcodes) -- but that should be a pretty narrow use case if anyone ever actually does it.